pkg/replicate/scheme.go

```diff
@@ -59,6 +59,7 @@ func NewBlockFilter(
 
 // Filter return true if block is non-compacted and matches selector.
 func (bf *BlockFilter) Filter(b *metadata.Meta) bool {
+	// TODO(bwplotka): Allow injecting custom labels as shipper does.
```

Comment:
I don't think this todo should be here tho, it should be in replicator. As replicator should be able to inject labels like shipper does and not this block filter. This block filter is the last filter that will remove blocks without labels. So labels should be injected before they reach this filter.

Reply:
Good catch! I agree it should be in the replicator.